compiler/src/dmd/func.d

```diff
@@ -3008,7 +3008,7 @@ extern (D) int overloadApply(Dsymbol fstart, scope int delegate(Dsymbol) dg, Sco
             else if (auto os = d.isOverloadSet())
             {
                 foreach (ds; os.a)
-                    if (int r = dg(ds))
+                    if (int r = overloadApplyRecurse(ds, dg, sc))
```

Comment:
Isn't this change going to impact other uses of this function?

Reply:
Yes, maybe? I don't understand the overload logic here. It seems that this function could stuff an entire overload set into the callback, which then presumably split it up further. I suspect there isn't a *semantic* difference here between "the callback splits up an overload set" and "the callback is called multiple times with the functions in an overload set". And the fact that buildkite is green seems to bear this out.

Reply:
> I'm not sure why this would be useful, as the point of having overload sets is to not confuse the members of one with the members of another, i.e. the anti-hijacking overload rules. With this change, functions that just happen to have the same name but are in different modules with utterly different uses will be lumped together.

Oh, is that how it works? If you have three modules, `a, b, c`, where `c` imports `a` and `b`, and all three have a `foo` respective, then there's three overload sets for "foo"? Do these all resolve on a lookup to "foo" and then get filtered down?